docs/_docs/modules.md

```diff
@@ -0,0 +1,90 @@
+
+| Function              | Class                         |
+| --------------------- | ----------------------------- |
+| @Design               | @DesignProperty               |
+| @DesignVariant        | @DesignVariantProperty        |
+| @DesignContentTypes   | @DesignContentTypesProperty   |
+| @DesignPreviewContent | @DesignPreviewContentProperty |
+
+Note that once this bug has been fixed, all of the `@Design*Property` annotations will be deprecated for at least one release, and then later removed, in favor of the analogous annotations used for functions.
+
+### Example:
+```kotlin
+@DesignModuleClass
+class TextModule(
+    @DesignProperty(node = "#text") val text: String,
+    @DesignProperty(node = "#replace")
+    val replaceNode: @Composable (ComponentReplacementContext) -> Unit,
+    @DesignVariantProperty(property = "#text-style") val textStyle: TextStyle,
+    @DesignContentTypesProperty(nodes = ["#button", "#header"])
+    @DesignPreviewContentProperty(
+        name = "Buttons",
+        nodes = [PreviewNode(3, "#button")]
+    )
+    val content: ListContent,
+)
+```
+
+## Nested Modules
+Module classes can include other modules. The resulting set of customizations will contain all customizations from the current class as well as all nested module classes. Use the `@DesignModuleProperty` annotation to achieve this, using the class name directly as the type of the `@DesignModuleProperty` class property.
+
+### Example:
+```kotlin
+@DesignModuleClass
+class TextModuleCombined(
+    @DesignProperty(node = "#name") val name: String,
+    @DesignModuleProperty val textProperties: TextModule,
+)
+```
+
+## Usage in a function
+Module classes by themselves are not useful until used as a parameter in a `@DesignComponent` function. Use the `@DesignModule` annotation for parameters in such a function. The generated function will combine any customizations from `@Design` and `@DesignVariant` parameters in the function with all the embedded customizations from parameters that are modules.
+
+### Example:
+```kotlin
+interface ModuleExample {
+    @DesignComponent(node = "#stage")
+    fun Main(
+        @Design(node = "#header") header: String,
+        @DesignModule moduleCustomizations: TextModuleCombined,
+    )
+}
+```
+
+To call the generated function, an instance of TextModuleCombined must be passed into the `moduleCustomizations` parameter. This can be constructed normally since it uses the developer defined class directly, not a generated one.
```

Comment:
Keep this line generic and if you feel you need a bit more clarification you can add comments  into the code block below.

```suggestion
To call the generated function, an instance of the design module class must be passed into the design module parameter. This can be constructed normally since it uses the developer-defined class directly, not a generated one.
```

Reply:
This is a clarity issue. Throughout the rest of the file you're describing how to use the annotations, but in this line you describe how to use the examples that you'd created. Someone reading the documentation but not paying close attention to the examples (or were reading through the main text first) would then need to go see what `TextModuleCombined` and `moduleCustomizations` are to understand that you're saying to add an instance of the design module class to the parameter.

Reply:
It's describing the examples because it is under the `Example` subheader. Do you want me to just move this into the code as a comment?

Reply:
Ok, I didn't realize that. This is the only place that you mention how to use the generated function so I assumed that this line was supposed to be part of the general documentation, not something explaining the example.

So here's the underlying problem, and this is what I was getting at in my first comment saying "please add an example of how to use it". The "Defining a module class" section tells you how to create a module and "Nested Modules" shows you how to add them to another `DesignModule`. Then you have "Usage in a function". The first sentance doesn't actually tell you how to use it in a function, it tells you how you can add the `DesignModule` to a `DesignComponent`, and this is what the first code block shows. Then you say "the generated function will combine any customizations...", which describes what the generated code will look like, but not how you would use the "use a DesignModule" in a generated function. It won't be obvious what this means, it certainly isn't obvious to me.


So what I expect from a section called "Usage in a function" is "how do write the correct syntax that writes the values for the `DesignModule` in the `DesignComponent` that uses it". This is what I thought the "to call the generated function" line was doing, explaining how to write the code, so I assumed you had made a mistake by using the names from the example code.


So let me sum that all up and rephrase the review comments with some additional context: The "Usage in a function" section doesn't describe how to use it in a function. Please rename it to "Adding to a `DesignComponent`". Then add a new section named "Usage in a DesignCompose function" that explains how to assign values to the `DesignModule` customizations in the generated DesignCompose functions in generic terms, just like how the other sections are written using generic terms. You will probably want to include the last example code block in this new section. Also, make sure that all text that refers to example code clearly says so by prefacing the text with something like "in the below example" or by including the text as comments within the example code.

Reply:
I called it "Usage in a function" because the `@DesignComponent` annotation is for a function, so you are using it for a function. It was supposed to encapsulate both the declaration of the function and the calling of the generated function. I've updated per your suggestion... I think? Though I have changed the section titles.